### DISCUSSION TITLE
Outer/inner borders

### DISCUSSION THREAD

**Discussion by superveryfastcar** [hyprwm/Hyprland#1499] [General]
Afaik, this is not a feature in Hyprland. I'd love to have something like a "col.inner_border" and a "col.outer_border". This would both be aesthetical and practical, allowing for example a more subtle an aesthetical way of showing a windows focus state. I'd create a pr myself, if i knew how to code. 

All the best wishes to the mainters, a wonderful project this is.

**Comment by adamnejm** [comment]
Having a one pixel border around Firefox prevents from interacting with the tabs, without the border you can just yeet the mouse towards the top of the screen while only being concerned by its horizontal position.

Rendering a non-interactable border over the Firefox would communicate which client is focused and still allow for the interaction.
This would be a big UX improvement.

**Reply by superveryfastcar** [reply]
I haven't tried it yet since I'm not using Hyprland rn, but theres a plugin called borders++ which I think basically does what this feature request wants. It allows extra window borders. Try it and see if it's what you're looking for.

Don't know if it changes interactability, but it's worth a shot!

**Reply by adamnejm** [reply]
Sorry for the late response. You were on the right track, setting the default's border to a negative size will offset the client. Then adding one additional border would offset it back and draw a border on top of it.
```haskell
general {
	border_size = -1
}

plugin {
    borders-plus-plus {
        add_borders = 1
        border_size_1 = 1
        col.border_1 = rgb(ff0000)
    }
}
```
Unfortunately like you said, it's static and I don't think there's a way to modify the color dynamically to indicate a focused application, even with external scripting.

**Reply by superveryfastcar** [reply]
That's unfortunate. I'd have imagined you could do some magic with hyprctl, getting active/inactive status and listing to window events. It seems good enough for me, though i would have liked to dynamically set border color and size.
I'll get on and try it out when i find some extra time.
